File: 830-largest-triangle-area/largest-triangle-area.java
class Solution {
    public double largestTriangleArea(int[][] points) {
        int n = points.length;
        double max_area = 0.0;

        for (int i = 0; i < n; i++)
            for (int j = i+1; j < n; j++)
                for (int k = j+1; k < n; k++) {
                    int x1 = points[i][0];
                    int y1 = points[i][1];
                    int x2 = points[j][0];
                    int y2 = points[j][1];
                    int x3 = points[k][0];
                    int y3 = points[k][1];

                    double curr_area = 0.5 * Math.abs(x1 * (y2-y3) + x2 * (y3-y1) + x3 * (y1-y2));
                    if (curr_area > max_area) max_area = curr_area;
                }

        return max_area;
    }
}

File: 830-largest-triangle-area/largest-triangle-area.py
class Solution:
    def largestTriangleArea(self, points: List[List[int]]) -> float:
        max_area = 0
        n = len(points)

        for i in range(n):
            for j in range(i+1, n):
                for k in range(j+1, n):
                    x1, y1 = points[i][0], points[i][1]
                    x2, y2 = points[j][0], points[j][1]
                    x3, y3 = points[k][0], points[k][1]

                    new_area = 0.5 * abs(x1*(y2-y3) + x2*(y3-y1) + x3*(y1-y2))
                    max_area = max(new_area, max_area)

        return max_area